Eb 1737 refactor sdk

vinay3206 opened the pull request.

maartenhemmes commented on src/data/notification/__test__/__snapshots__/notification.test.js.snap:
```diff
@@ -1,27 +1,3 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
```
@vinay3206 why is the snapshot empty?

vinay3206 replied:
@maartenhemmes snapshot got updated because response is empty.

Actually, u can not always expect the same notifications in the response so snapshot needs to be kept updated

maartenhemmes commented on src/data/Endpoint.js:
```diff
@@ -25,22 +25,22 @@ const endpoints = {
-    GROUPING_LOCATION_COUNT: `${laravelUrlPrefix}/${version}/customer/grouping-location/total`,
+    ORDER: `${laravelUrlPrefix}/${version}/customer/order`,
+    ORDER_COUNT: `${laravelUrlPrefix}/${version}/customer/order/total`,
     GROUPING_BATCH_PROGRESSION: `${laravelUrlPrefix}/${version}/customer/grouping-batch/show-recent-pending`,
```
@vinay3206 shouldn't this be renamed (GROUPING_BATCH_PROGRESSION) ?

maartenhemmes replied:
@vondich @kianxiongfoo what's the reason we left this out from the refactoring?

vinay3206 commented on src/data/customer/__test__/order.test.js:
```diff
@@ -290,26 +290,42 @@ describe('Convert Ids into CSV string', () => {
-      CONFIG.token
-    );
-    expect('data' in response).toBeTruthy();
-    expect(response.data).toBeTruthy();
```
yes because those records does not exist in the dynamo db now

vinay3206 commented on src/data/customer/__test__/order.test.js:
```diff
@@ -290,26 +290,42 @@ describe('Convert Ids into CSV string', () => {
-    expect('data' in response).toBeTruthy();
-    expect(response.data).toBeTruthy();
+    try {
+      const response = await removeOrderErrorRecordsAsync(
```
We have hardcoded Ids to delete. They are not present in the dynamo db right now in this environment

vinay3206 commented on src/data/notification/__test__/notification.test.js:
```diff
@@ -20,22 +20,39 @@ describe('Test for retrieving all notifications', () => {
-    const response = getNotificationsAsync(false, 1, token.accessToken);
-    const notifications = await response;
-    expect('data' in notifications).toBeTruthy();
-    expect(notifications.data).toMatchSnapshot();
```
If no notifications are there then API always returns 404.

Which is the case with pre-test environement

vinay3206 commented on src/data/customer/Job.js:
```diff
@@ -76,3 +76,98 @@ export const getRecommendedJobsAsync = async (filterObject = {}, token) => {
+}
+
+/**
+ * Categories Customer Orders
```
@maartenhemmes  this function is used by functions on the dashboard. so it is job related.

I have updated it now

maartenhemmes replied:
ok, but then you should call it Job in the description?

vinay3206 commented on src/data/customer/Job.js:
```diff
@@ -76,3 +76,98 @@ export const getRecommendedJobsAsync = async (filterObject = {}, token) => {
+ //TODO: needs unit testing
+ */
+function calculateCustomerJobsCounts(data) {
+  let orders = categoriesCustomerJobs(data);
```
@maartenhemmes  its for job only, I just updated the name before. Now i have updated the internal functions too.

vinay3206 commented on src/algo/__test__/routing.test.js:
```diff
@@ -25,9 +25,7 @@ describe('Call route optimization endpoints', () => {
-        errorMessage: [
-          { key: '0', messages: 'Route Setting does not belong to you' },
-        ],
+        errorMessage: ['Route Setting does not belong to you'],
```
added the key again!

maartenhemmes replied:
@vinay3206 probably good to convert as you are converting all your return values. But I don't see it here yet?

maartenhemmes replied:
I saw you are not doing it everywhere though, might be good to be consistent about this. @vinay3206

vinay3206 commented on src/data/customer/Job.js:
```diff
@@ -76,3 +76,98 @@ export const getRecommendedJobsAsync = async (filterObject = {}, token) => {
+ * @param {int} customerId
+ * @param {string} token
+ * @return {object} Promise resolve/reject
+ */
```
I have putted that comment back

vinay3206 commented on src/data/customer/Job.js:
```diff
@@ -76,3 +76,98 @@ export const getRecommendedJobsAsync = async (filterObject = {}, token) => {
+
+/**
+ * Get Jobs with filters for Dashboard
+ * @param {object} filterObject # {pickupDate (mandatory), routeStatusIds, includeOrders, limit, offset}
```
updated

vinay3206 commented on src/data/customer/Order.js:
```diff
@@ -522,34 +460,33 @@ export const updateAndTruncateOrderErrorsAsync = async (
-    await removeOrderErrorRecordsAsync(errorIds, token);
+    if (editResponse) {
+      await removeOrderErrorRecordsAsync(errorIds, token);
+    }
```
Removed unnecessary if

vinay3206 commented on src/data/customer/Order.js:
```diff
@@ -885,39 +816,6 @@ export const getUpdatedJobLiveData = (
   }
```
Moved and renamed.

vinay3206 commented on src/data/customer/Order.js:
```diff
@@ -885,39 +816,6 @@ export const getUpdatedJobLiveData = (
- * @param {object} data
- * @return {object} data # retrun count of data object
- //TODO: needs unit testing
- */
```
renamed

vinay3206 commented on src/data/notification/__test__/notification.test.js:
```diff
@@ -20,22 +20,39 @@ describe('Test for retrieving all notifications', () => {
-    expect('data' in response.data).toBeTruthy();
-    expect(response.data).toEqual(deleteResponse);
-    expect(true).toBeTruthy();
+    try {
```
@maartenhemmes because notification id we are using to delete is not there in pre-test environment

vinay3206 commented:
@maartenhemmes some tests are failing right now because of timeout.

maartenhemmes commented on src/algo/__test__/routing.test.js:
```diff
@@ -26,7 +26,10 @@ describe('Call route optimization endpoints', () => {
-          { key: '0', messages: 'Route Setting does not belong to you' },
+          {
+            key: 'route_setting_id',
+            messages: ['Route Setting does not belong to you'],
```
snakecase or camel, you will need to choose @vinay3206

vinay3206 replied:
updated keys to camelCase

maartenhemmes commented:
@vinay3206 unit tests are failing again

maartenhemmes commented:
@vinay3206 unit tests are failing again

vinay3206 commented:
@maartenhemmes 
its because of some more changes from the API as nin told me today

- Renamed location_data to order_data
- Changed type of location_data / order_data from JSON-encoded string to JSON object in request

Will update these 2mmrow

maartenhemmes commented on src/data/Endpoint.js:
```diff
@@ -1,6 +1,6 @@
 const urlPrefix =
-  'https://b2gp6sn1o4.execute-api.ap-southeast-1.amazonaws.com/staging';
-const laravelUrlPrefix = 'https://api-test.carpal.me';
+  'https://b2gp6sn1o4.execute-api.ap-southeast-1.amazonaws.com/pre-test';
```
before we merge it we should change this back
